src/dynapi/SDL_dynapi.c

```diff
@@ -152,15 +152,31 @@ static void SDL_InitDynamicAPI(void);
         return result;                                                                                                                    \
     }                                                                                                                                     \
-    _static bool SDLCALL SDL_RenderDebugTextF##name(SDL_Renderer *renderer, float x, float y, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) \
+    _static bool SDLCALL SDL_RenderDebugText##name(SDL_Renderer *renderer, float x, float y, SDL_PRINTF_FORMAT_STRING const char *fmt, ...) \
     {                                                                                                                                     \
-        bool result;                                                                                                                      \
+        char buf[128], *str = buf;                                                                                                        \
+        int result;                                                                                                                       \
         va_list ap;                                                                                                                       \
         initcall;                                                                                                                         \
         va_start(ap, fmt);                                                                                                                \
-        result = jump_table.SDL_RenderDebugTextV(renderer, x, y, fmt, ap);                                                                \
+        result = jump_table.SDL_vsnprintf(buf, sizeof(buf), fmt, ap);                                                                     \
         va_end(ap);                                                                                                                       \
-        return result;                                                                                                                    \
+        if (result >= 0 && (size_t)result >= sizeof(buf)) {                                                                               \
+            const size_t len = (size_t)result + 1;                                                                                        \
+            str = (char *)jump_table.SDL_malloc(len);                                                                                     \
+            if (str) {                                                                                                                    \
+                va_start(ap, fmt);                                                                                                        \
+                result = jump_table.SDL_vsnprintf(str, len, fmt, ap);                                                                     \
```

Comment:
Maybe use SDL_vasprintf() instead?

Reply:
This was copy/pasted from the SDL_SetError wrapper code, but yes, that's a good idea.

Reply:
Fixed in e915a35345b690e9a3bbd2a58ffd9da850db8cf1.